runtime/browser/xwalk_runner.cc

```diff
@@ -163,6 +169,48 @@ void XWalkRunner::OnRenderProcessWillLaunch(content::RenderProcessHost* host) {
       std::move(runtime_variables));
 }
 
+void XWalkRunner::InitializeEnvironmentVariablesForGoogleAPIs(
+    content::RenderProcessHost* host) {
+#if defined (OS_WIN)
+  application::Application* app =
+      app_system()->application_service()->
+      GetApplicationByRenderHostID(host->GetID());
+  if (!app)
+    return;
+  std::string app_name;
+  if (!app->data()->GetManifest()->GetString("xwalk_package_id", &app_name))
+    return;
+  std::string vendor =
+      app_name.substr(0, app_name.find_last_of("."));
+  std::wstring registry_path(L"SOFTWARE\\");
+  // Assuming xwalk_package_id is ASCII.
```

Comment:
http://stackoverflow.com/questions/2573834/c-convert-string-or-char-to-wstring-or-wchar-t
I would try std::wstringstream but you might prefer other ways..